src/client/game.cpp

```diff
@@ -1193,22 +1193,28 @@ void Game::run()
 
 	irr::core::dimension2d<u32> previous_screen_size(g_settings->getU16("screen_w"),
 		g_settings->getU16("screen_h"));
+	bool was_window_maximized = g_settings->getBool("window_maximized");
 
 	while (m_rendering_engine->run()
 			&& !(*kill || g_gamecallback->shutdown_requested
 			|| (server && server->isShutdownRequested()))) {
 
 		const irr::core::dimension2d<u32> &current_screen_size =
 			m_rendering_engine->get_video_driver()->getScreenSize();
+		const bool is_window_maximized = RenderingEngine::get_raw_device()
+				->isWindowMaximized();
 		// Verify if window size has changed and save it if it's the case
-		// Ensure evaluating settings->getBool after verifying screensize
-		// First condition is cheaper
-		if (previous_screen_size != current_screen_size &&
-				current_screen_size != irr::core::dimension2d<u32>(0,0) &&
-				g_settings->getBool("autosave_screensize")) {
-			g_settings->setU16("screen_w", current_screen_size.Width);
-			g_settings->setU16("screen_h", current_screen_size.Height);
+		if ((was_window_maximized != is_window_maximized
+					|| previous_screen_size != current_screen_size)
+				&& g_settings->getBool("autosave_screensize")) {
+			// Don't replace good value with (0, 0)
+			if (current_screen_size != irr::core::dimension2d<u32>(0, 0)) {
+				g_settings->setU16("screen_w", current_screen_size.Width);
+				g_settings->setU16("screen_h", current_screen_size.Height);
+			}
 			previous_screen_size = current_screen_size;
+			g_settings->setBool("window_maximized", is_window_maximized);
+			was_window_maximized = is_window_maximized;
 		}
```

Comment:
Wouldn't it suffice to update this after the loop?

Reply:
It would, probably.
But there might be differences if you're running multiple minetest instances, or if the client crashes. (Idk when and how often minetest writes its settings to file.)
As performance-wise it makes no difference, I'd like to leave it like before unless there are good arguments for change.

Reply:
AFAIK, the settings are only written at exit, hence one instance will overwrite the other anyway. Upon client crash, unsaved settings (anything before the write call) are lost. In this case it would not be dramatic.

Reply:
Done.